### PR TITLE
WIP: Riak KV 1329 tests

### PR DESCRIPTION
**Pull request by lukebakken**
Adding tests for 2i exact match scenarios

See basho/riak_kv#1329

### CODE REVIEW

File: tests/verify_2i_returnterms.erl [hunk 1]
```diff
@@ -38,15 +40,20 @@ confirm() ->
...
     %% Bucket, key, and index_eq queries should ignore `return_terms'
-    ExpectedKeys = lists:sort([int_to_key(N) || N <- lists:seq(0, 200)]),
+    ExpectedKeys = lists:sort([int_to_key(N) || N <- lists:seq(0, 202)]),
     assertEqual(RiakHttp, PBPid, ExpectedKeys, {<<"$key">>, int_to_key(0), int_to_key(999)}, ?Q_OPTS, keys),
     assertEqual(RiakHttp, PBPid, ExpectedKeys, { <<"$bucket">>, ?BUCKET}, ?Q_OPTS, keys),
 
     ExpectedFooKeys = lists:sort([int_to_key(N) || N <- lists:seq(101, 200)]),
     assertEqual(RiakHttp, PBPid, ExpectedFooKeys, {<<"field1_bin">>, ?FOO}, ?Q_OPTS, keys),
 
+    assertEqual(RiakHttp, PBPid, [int_to_key(201)], {<<"field1_bin">>, ?BAZ}, ?Q_OPTS, keys),
```
Comment (JeetKunDoug) [inline tests/verify_2i_returnterms.erl]: Please make sure to put some comments around/before the newly added expectations so we know why they were added and what they are testing.

**Comment by javajolt** [comment]
@lukebakken Would you mind doing this on the `riak/2.0` branch, assuming this is appropriate for that version too?  `master` is the end of the line, not the start these days.  The merge order today is `riak/2.0` -> `riak/2.1` -> `riak-ts_develop-1.2`.  We should update master from `riak/2.1` soon, too.

**Comment by JeetKunDoug** [comment]
Funny - see #977

**Comment by lukebakken** [comment]
@javajolt - I'll do that when I get a chance.

**Comment by lukebakken** [comment]
@javajolt @JeetKunDoug - I rebased on `riak/2.0` and opened #982

**Pull request closed without merging**